### PR TITLE
src/sage/combinat: Doctest cosmetics

### PR DESCRIPTION
**Pull request by mkoeppe**
Standard reformatting of doctests and their outputs

Split out from https://github.com/sagemath/sage/pull/35095

### :memo: Checklist

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/combinat/path_tableaux/semistandard.py [hunk 1]
```diff
@@ -12,13 +12,15 @@
 
 EXAMPLES::
 
-    sage: pt = path_tableaux.SemistandardPathTableau([[],[3],[3,2],[3,3,1],[3,3,2,1],[4,3,3,1,0]])
+    sage: pt = path_tableaux.SemistandardPathTableau([[],[3],[3,2],[3,3,1],
```
Comment (dcoudert) [inline src/sage/combinat/path_tableaux/semistandard.py]: you could add missing spaces inside the lists

File: src/sage/combinat/path_tableaux/semistandard.py [hunk 2]
```diff
@@ -387,7 +398,8 @@ def to_tableau(self):
 
         EXAMPLES::
 
-            sage: pt = path_tableaux.SemistandardPathTableau([[],[3],[3,2],[3,3,1],[3,3,2,1],[4,3,3,1,0]])
+            sage: pt = path_tableaux.SemistandardPathTableau([[],[3],[3,2],[3,3,1],
```
Comment (dcoudert) [inline src/sage/combinat/path_tableaux/semistandard.py]: on the way, you could add missing spaces in lists

File: src/sage/combinat/permutation.py [hunk 3]
```diff
@@ -5508,11 +5511,11 @@ def rewind(L, n):
 
     def has_nth_root(self, n) -> bool:
         r"""
-        Decide if ``self`` has n-th roots.
+        Decide if ``self`` has `n`-th roots.
```
Comment (dcoudert) [inline src/sage/combinat/permutation.py]: `Decide` or `Check` ?

File: src/sage/combinat/posets/poset_examples.py [hunk 4]
```diff
@@ -969,9 +969,9 @@ def SSTPoset(s, f=None):
 
         INPUT:
 
-        - ``s`` - shape of the tableaux
+        - ``s`` -- shape of the tableaux
 
-        - ``f`` - maximum fill number.  This is an optional
+        - ``f`` -- maximum fill number.  This is an optional
```
Comment (dcoudert) [inline src/sage/combinat/posets/poset_examples.py]: could use the standard form
```py
- ``f`` -- integer (default: ``None``); specify the maximum fill number.
  By default (``None``), the method uses the number of cells in the shape.
```

File: src/sage/combinat/tiling.py [hunk 5]
```diff
@@ -1377,8 +1393,8 @@ def show3d(self, size=1):
 
         INPUT:
 
-        - ``self`` - a polyomino of dimension 3
-        - ``size`` - number (optional, default: ``1``), the size of each
+        - ``self`` -- a polyomino of dimension 3
+        - ``size`` -- number (optional, default: ``1``), the size of each
```
Comment (dcoudert) [inline src/sage/combinat/tiling.py:1397]: we could remove the term `optional`. We have it many times for parameters with default value.

File: src/sage/combinat/path_tableaux/semistandard.py (hunk 2 above)
Comment (mkoeppe) [reply src/sage/combinat/path_tableaux/semistandard.py]: Done in 0f7b0307a3e

File: src/sage/combinat/permutation.py (hunk 3 above)
Comment (mkoeppe) [reply src/sage/combinat/permutation.py]: Done in ebea641dcbc

File: src/sage/combinat/posets/poset_examples.py (hunk 4 above)
Comment (mkoeppe) [reply src/sage/combinat/posets/poset_examples.py]: Done as part of 873c7801028

File: src/sage/combinat/tiling.py (hunk 5 above)
Comment (mkoeppe) [reply src/sage/combinat/tiling.py:1397]: I've left the "optional" as is, too many changes in this PR already

**Review by dcoudert — APPROVED** [review]
LGTM

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**